api/v1alpha2/cluster_types.go

```diff
@@ -48,11 +48,11 @@ type ClusterSpec struct {
 type ClusterNetworkingConfig struct {
 	// The network ranges from which service VIPs are allocated.
 	// +optional
-	Services NetworkRanges `json:"services,omitempty"`
+	Services *NetworkRanges `json:"services,omitempty"`
```

Comment:
These are already slices, any reason we need a pointer?

Reply:
NetworkRanges is actually a struct, not a aliased type. The name threw me off as well, after the optional changes you brought in, validation failed on the inner `cidrBlocks` because that field isn't optional, but the struct is always populated with a non-nil value.